core/modules/image/image.field.inc

```diff
@@ -292,13 +292,18 @@ function image_field_presave($entity_type, $entity, $field, $instance, $langcode
 
   // Determine the dimensions if necessary.
   foreach ($items as &$item) {
-    if (!isset($item['width']) || !isset($item['height'])) {
+    if (empty($item['width']) || empty($item['height'])) {
       $info = image_get_info(file_load($item['fid'])->uri);
 
       if (is_array($info)) {
         $item['width'] = $info['width'];
         $item['height'] = $info['height'];
       }
+      else {
```

Comment:
You could put the variable defaulting to NULL before the if statement. That way it's clear they'll always be NULL unless width and height are available.

Reply:
good idea. done!

Reply:
Better. My suggestion also applies to the first part of this patch as well. The setting of NULL can be above the if statement.

Reply:
of course :) done!